adapters/amqp-vertx-base/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -600,6 +601,11 @@ protected void handleRemoteReceiverOpen(final ProtonConnection conn, final Proto
                 try {
                     final SpanContext spanContext = TracingHelper.extractSpanContext(tracer, message);
                     final Span msgSpan = newSpan("upload message", authenticatedDevice, traceSamplingPriority, spanContext);
+                    HonoProtonHelper.onReceivedMessageDeliveryUpdatedFromRemote(delivery, d -> {
+                        log.debug("got unexpected disposition update for message received from device [remote state: {}, container: {}, {}]",
+                                delivery.getRemoteState(), conn.getRemoteContainer(), authenticatedDevice);
+                        TracingHelper.logError(msgSpan, "got unexpected disposition from device [remote state: " + delivery.getRemoteState() + "]");
```

Comment:
if we consider this to be an error, shouldn't we then also prevent the message from being processed? Otherwise I would opt for not logging this as an error but simply as an information.

Reply:
Message processing is most likely already in progress at the point the handler is called. 
A way to let that delivery update from the remote peer influence the message processing would be checking the `remotelySettled` flag of the delivery as part of message processing - as done in `ProtonBasedCommandContext`. But that needs to be implemented separately. Therefore having an error logged in the span in any case is something I would prefer here.

Reply:
But there is no error, is there?

Reply:
Well, ok, not necessarily. I'll just add a span log entry.